src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -144,12 +144,11 @@ const StudentApplication = ({ userProfile, authentication }) => {
   /**
    * Check whether a given applicant is valid for this current application
    *
-   * @async
    * @function isApplicantValid
    * @param {ApartmentApplicant} applicant The applicant to be checked
    * @return {Boolean} True if valid, otherwise false
    */
-  const isApplicantValid = async (applicant) => {
+  const isApplicantValid = (applicant) => {
```

Comment:
@EjPlatzer  I am guessing that the async behavior might have caused problems on the line with 
``` javascript
... else if (isApplicantValid(newApplicantObject)) { ...
```
Which was being handled as "true" in cases that clear should have been false,

I am not entirely familiar with the details of how async functions behave, so let me know your thoughts

Reply:
Yes, if you were to call it like you mentioned, then that would definitely cause issues. An `async` function always returns a Promise, which will resolve to the value returned from the function once the function finishes executing. So in the case you mentioned, `if (isApplicantValid(newApplicantObject))` is equivalent to `if (Promise)` which evaluates to `if (true)`.